### PR TITLE
Add Free Music Archive to external sources in the frontend

### LINKED ISSUE #2486: Add Free Music Archive to the "External Sources" frontend feature
## Problem

The FMA is a lovely source of CC-licensed and public domain audio files. We could easily add them to our "External sources" functionality:

`https://freemusicarchive.org/search?quicksearch={term}`

### PR DESCRIPTION
**Pull request by alt-shreya**
## Fixes

Fixes #2486  by @zackkrida 

## Description

Added  `Free Music Archive` to the inbuilt `External sources` frontend functionality.

Screenshot from 2024-02-27 18-03-06

image_screenshot_fma_test
Screenshot from 2024-02-27 18-03-39

## Testing Instructions

_Optional:_ 
1. Run `just frontend/run dev`
2. Open an audio search page
3. Scroll down to the "External sources" button

The popup should contain "Free Music Archive", and clicking on it should give the results of the query on Openverse website.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by alt-shreya** [comment]
@AetherUnbound tagging you here as you assigned this issue to me :sparkles:

**Review by obulat — CHANGES_REQUESTED** [review]
Hi @alt-shreya,
This is a great start, thank you for the screenshots in the PR description!
The tests are failing because the snapshots are now outdated. You need to update the snapshots, here are the instructions on how to do that:  https://docs.openverse.org/frontend/guides/test.html#updating-snapshots

Comment here if you need any help, I'll be happy to do so or to update the snapshots for you if needed.

**Comment by obulat** [comment]
@alt-shreya, I am going to draft this PR while the changes are being made to prevent Openverse bot from pinging for reviews. Please undraft it when you add the snapshots.

**Comment by alt-shreya** [comment]
> Hi @alt-shreya, This is a great start, thank you for the screenshots in the PR description! The tests are failing because the snapshots are now outdated. You need to update the snapshots, here are the instructions on how to do that: https://docs.openverse.org/frontend/guides/test.html#updating-snapshots
> 
> Comment here if you need any help, I'll be happy to do so or to update the snapshots for you if needed.

@obulat I've updated the snapshots in my most recent commits. I only updated for `visual-regression` as these were the tests that were failing. Please let me know if you have more feedback!

File: frontend/src/utils/get-additional-sources.ts [hunk 1]
```diff
@@ -235,6 +235,16 @@ const additionalSourceBuilders: AdditionalSourceBuilder[] = [
       }
     },
   },
+
```
Comment (AetherUnbound) [inline frontend/src/utils/get-additional-sources.ts:238]:
```suggestion
```

**Review by AetherUnbound — APPROVED** [review]
Looks great, thanks for the contribution! 😄 One nit for a newline that doesn't appear necessary.

**Review by obulat — APPROVED** [review]
Awesome, thank you for your contribution, @alt-shreya!

**Comment by alt-shreya** [comment]
@obulat @AetherUnbound 
Thanks for all your help! I had a great time contributing! Looking forward to making more PRs :smile: :tada:

**Pull request merged**